### PR TITLE
Fix root

### PR DESCRIPTION
**Pull request by euclideansphere**
Fixes the root login issue (sudo is not a thing on a minimal rasperry pi, we should just run as root)
Prompts the user for the webhook url for easier setup

### CODE REVIEW

**Comment by mattmahn** [comment]
Couple issues:
1. Fix indentation at [line 12 of install.sh](https://github.com/MSOE-Supermileage/crust/pull/6/files#diff-3fbb47e318cd8802bd325e7da9aaabe8R12)
2. [This `sed` command](https://github.com/MSOE-Supermileage/crust/pull/6/files#diff-3fbb47e318cd8802bd325e7da9aaabe8R78) will not do anything since there is no string matching `$REPLACEME` in config.json.
3. Also, that [`sed` command](https://github.com/MSOE-Supermileage/crust/pull/6/files#diff-3fbb47e318cd8802bd325e7da9aaabe8R78) could be more clearly written using double quotes:
   
   ``` sh
   sed -i "s|$REPLACEME|$URL|g" $DIRNAME/config.json
   ```

Just a thought: instead of prompting for the webhook URL, we could simply accept it as a parameter when running install.sh. However, this is not crucial for our purposes.

**Comment by euclideansphere** [comment]
The sed command needs single quotes to interpret as a literal, if I recall correclty
http://stackoverflow.com/questions/6697753/difference-between-single-and-double-quotes-in-bash

also I'll modify to just accept as a param, I like that better.

**Comment by mattmahn** [comment]
I've done some [testing](https://gist.github.com/mattmahn/1a1d94e3eb1becb41a2e), and there seems to be some issues. The issue seems to be that `|` cannot be used to delimit the regexp-replacement pair, so we need to use `/`, but that occurs in both `$REPLACEME` and `$URL` thus sed attempts the parse the characters after `/` in the URLs as options.

To fix this, while still using sed, one way is to specify each of the three parts of the webhook URL. For example:

``` json
{
    "webhook-url": "https://hooks.slack.com/services/T00/B00/XXX"
}
```

``` sh
$ echo T_URL
T56415H32R0564
$ echo B_URL
B0564FS515
$ echo X_URL
FSDfSdfsdFsdFSDaF
$ sed -i "s/T00/$T_URL/" config.json
$ sed -i "s/B00/$B_URL/" config.json
$ sed -i "s/XXX/$X_URL/" config.json
```

However, this is quite cumbersome to separately handle each of the 3 parts, so I think it may be better to stick with the copy+paste method.

**Comment by euclideansphere** [comment]
try my latest commit

you can put in the url as a command line param

**Comment by mattmahn** [comment]
That doesn't work either. It seems that `|` doesn't work, even when escaped as `\|` which the [sed documentation says](https://www.gnu.org/software/sed/manual/sed.html#The-_0022s_0022-Command) will work. But when using `/`, that character is in the URL, so it ends the s command and attempts to parse the characters following the first `/` as flags, which it cannot do since they are invalid characters.

File: crust.py [hunk 1]
```diff
@@ -60,3 +69,4 @@ def main():
 
 if __name__ == '__main__':
     main()
+
```
Comment (euclideansphere) [inline crust.py:71]: you know what houndci maybe I want a damn line at the end of my file because it's in the damn PEP8 standard

**Comment by mattmahn** [comment]
When [adding the URL to config.conf](https://github.com/MSOE-Supermileage/crust/blob/fix_root/install.sh#L61), you forgot to assign the url to `webhook-url`.
Also, [this code block](https://github.com/MSOE-Supermileage/crust/blob/fix_root/install.sh#L76) doesn't seem very useful were it is now. It should probably go to line 5 or 11 instead.

I'll try running on a Pi tonight.

P.S. I'm working on getting Travis CI setup, so I turned it on for the repo. We can ignore its errors for now...

**Pull request closed without merging**